src/persist-client/src/internal/machine.rs

```diff
@@ -171,6 +171,7 @@ where
         purpose: &str,
         lease_duration: Duration,
         heartbeat_timestamp_ms: u64,
+        use_critical_since: bool,
```

Comment:
I hate to be annoying, but maybe an enum is better than all the new `false` at callsites? 🙈

Reply:
This is true, but:
- hopefully we can shift the behaviour fast enough for this to not matter too much;
- the enum is slightly more annoying to use with dyncfg, which I expect to drive most of the migration.

If this ends up lingering, though, I'll make a note to follow up with this!